### ISSUE TITLE
pointrend 3.x empty roi bug

### ISSUE THREAD

**Issue by treasureqqm** [open-mmlab/mmdetection#9950]
在3.x版本的point_rend_roi_head.py的predict_mask函数中，没有考虑到同一个batch中，有些图像有bbox，而有些图像没有bbox，即mask_rois = mask_rois.split(num_mask_rois_per_img, 0)里面存在空box，最后会在assert rois.shape[0] > 0, 'RoI is a empty tensor.'出错

**Comment by JolyonWu** [comment]
是的我也遇到同样的问题

**Comment by wgd4510** [comment]
同样的问题，请问现在解决了吗

**Comment by treasureqqm** [comment]
> 同样的问题，请问现在解决了吗

解决了，不过不是很优雅，在mask_rois_i = mask_rois[i]下面加三行就行了：

**Comment by JohannesTheo** [comment]
Little late to the party, but here is how I fixed it to train pointrend with LSJ augmentation. In this case, images can occasionally have no assigned gt label resulting in the empty roi error.

In `PointRendRoIHead` `_get_fine_grained_point_feats` I add:

```python
if rois.shape[0] == 0:
    # return an 'empty' zero tensor with matching shapes
    return torch.zeros(size=(0, x[0].shape[1], rel_roi_points.shape[1]), device=rois.device, dtype=torch.float32)
```

and in `MaskPointHead` `loss_and_target` I add:

```
# This is similar to how other mask heads handle this case
if point_pred.size(0) == 0:
    loss_point = point_pred.sum()
else:
    # normal if else loss code from here
```
